### ISSUE TITLE
SEGFAULT on loading exodus mesh with more than one process

### ISSUE THREAD

**Issue by juan42** [firedrakeproject/firedrake#995]
Executing this code with "mpiexec -n 2 scripy.py"
```
from firedrake import *
mesh = Mesh('disk_out_ref.e')
```
fails with the following error
```
[1]PETSC ERROR: ------------------------------------------------------------------------
[1]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of range
[1]PETSC ERROR: Try option -start_in_debugger or -on_error_attach_debugger
[1]PETSC ERROR: or see http://www.mcs.anl.gov/petsc/documentation/faq.html#valgrind
[1]PETSC ERROR: or try http://valgrind.org on GNU/linux and Apple Mac OS X to find memory corruption errors
[1]PETSC ERROR: configure using --with-debugging=yes, recompile, link, and run 
[1]PETSC ERROR: to get more information on the crash.
application called MPI_Abort(MPI_COMM_WORLD, 59) - process 1
```
This happens to any type of exodus mesh i have available. The plain C version for DMPlex works fine.
A sample mesh is available [here](https://cloud.tf.uni-kiel.de/index.php/s/GliLS7LNZo1suKz).

**Comment by wence-** [comment]
The boundary marking code needs to be guarded.  Try this patch:

```diff
diff --git a/firedrake/mesh.py b/firedrake/mesh.py
index 0501d5c..a8a2b95 100644
--- a/firedrake/mesh.py
+++ b/firedrake/mesh.py
@@ -203,12 +203,13 @@ def _from_exodus(filename, comm):
     """
     plex = PETSc.DMPlex().createExodusFromFile(filename, comm=comm)
 
-    boundary_ids = plex.getLabelIdIS("Face Sets").getIndices()
-    plex.createLabel("boundary_ids")
-    for bid in boundary_ids:
-        faces = plex.getStratumIS("Face Sets", bid).getIndices()
-        for f in faces:
-            plex.setLabelValue("boundary_ids", f, bid)
+    if plex.hasLabel("Face Sets"):
+        boundary_ids = plex.getLabelIdIS("Face Sets").getIndices()
+        plex.createLabel("boundary_ids")
+        for bid in boundary_ids:
+            faces = plex.getStratumIS("Face Sets", bid).getIndices()
+            for f in faces:
+                plex.setLabelValue("boundary_ids", f, bid)
 
     return plex
 
```

**Comment by wence-** [comment]
This code needs tidied, it repeats the same stuff as in `_from_gmsh`.

**Comment by wence-** [comment]
More generally, I don't understand why we take face sets, and basically copy them into boundary_ids (the information in the two is the same).  But that's not relevant here.

**Comment by wence-** [comment]
Fixed in #997
